src/main/java/de/zalando/zmon/dataservice/data/MetricTiers.java

```diff
@@ -88,13 +88,12 @@ public boolean isMetricEnabled(int checkId) {
         if (ingestMaxCheckTier == 1) {
             return criticalChecks.contains(checkId);
         } else if (ingestMaxCheckTier == 2) {
-            boolean isWhitelisted = criticalChecks.contains(checkId) || importantChecks.contains(checkId);
-            boolean isSampled = sampledCheckTier != 2 || random.nextDouble() > sampledCheckRate;
-            return isWhitelisted && isSampled;
+            return (criticalChecks.contains(checkId) || importantChecks.contains(checkId)) // is whitelisted
+                    && (sampledCheckTier == 0 || sampledCheckTier == 3 || random.nextDouble() > sampledCheckRate); // AND is sampled
```

Comment:
I'm not quite sure about how it should work.
From the code it looks like if `ingestMaxCheckTier == 2` (we ingest only critical and important), we also check the sampling level. And we assume, that it is either switched off (`sampledCheckTier == 0`), or points to the level we need to sample. If the level is 3, we don't care, neither when it's switched off. But if `sampledCheckTier == 2`, we should flip a coin. 

What I do not understand, is what we are fixing here? The condition `sampledCheckTier != 2` is totally equal to `sampledCheckTier == 0 || sampledCheckTier == 3` if set of possible values is `[0,2,3]'. So if it didn't work before, it shouldn't work now. Or it's too late and I miss something obvious.

Reply:
You are correct in this case. We are fixing the case `ingestMaxCheckTier == 3`, which was broken. This seemed not to be broken , I've changed it to look aligned with the third one